src/Stack/BuildPlan.hs

```diff
@@ -638,7 +637,7 @@ shadowMiniBuildPlan :: MiniBuildPlan
 shadowMiniBuildPlan (MiniBuildPlan cv pkgs0) shadowed =
     (MiniBuildPlan cv $ Map.fromList met, Map.fromList unmet)
   where
-    pkgs1 = Map.difference pkgs0 $ Map.fromSet (\_ -> ()) shadowed
+    pkgs1 = Map.difference pkgs0 $ Map.fromSet const shadowed
```

Comment:
Can you please leave this one out?  This actually causes the type of this map to change.  `(\_ -> ())` isn't the same as `const`.  Personally, I don't agree with some of hlint's suggestions.  Using `const` is one thing I don't like.  Compare:

`const ()`
`\_ -> ()`

The lambda version is shorter and more straightforward and naturally scales:

`const (const ())`
`\_ _ -> ()`